smoke-tests/framework/src/main/java/com/microsoft/applicationinsights/smoketest/fakeingestion/MockedOtlpIngestionServer.java

```diff
@@ -0,0 +1,86 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.microsoft.applicationinsights.smoketest.fakeingestion;
+
+import static java.util.concurrent.TimeUnit.SECONDS;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.awaitility.Awaitility.await;
+import static org.mockserver.integration.ClientAndServer.startClientAndServer;
+import static org.mockserver.model.HttpRequest.request;
+import static org.mockserver.model.HttpResponse.response;
+import static org.mockserver.stop.Stop.stopQuietly;
+
+import com.google.protobuf.InvalidProtocolBufferException;
+import io.opentelemetry.proto.collector.metrics.v1.ExportMetricsServiceRequest;
+import io.opentelemetry.proto.metrics.v1.Metric;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Objects;
+import java.util.stream.Collectors;
+import javax.annotation.Nullable;
+import org.mockserver.integration.ClientAndServer;
+import org.mockserver.model.Body;
+import org.mockserver.model.HttpRequest;
+
+public class MockedOtlpIngestionServer {
+  static final int EXPORTER_ENDPOINT_PORT = 4318;
+  static ClientAndServer collectorServer;
+
+  @SuppressWarnings("SystemOut")
+  public void startServer() throws Exception {
+    System.out.println("Starting fake OTLP ingestion...");
+    collectorServer = startClientAndServer(EXPORTER_ENDPOINT_PORT);
+    collectorServer.when(request()).respond(response().withStatusCode(200));
+  }
+
+  @SuppressWarnings("SystemOut")
+  public void stopServer() throws Exception {
+    System.out.println("Stopping fake OTLP ingestion...");
+    stopQuietly(collectorServer);
+  }
+
+  @SuppressWarnings("PreferJavaTimeOverload")
+  public void verify() {
+    await()
+        .atMost(60, SECONDS)
+        .untilAsserted(
+            () -> {
+              HttpRequest[] requests = collectorServer.retrieveRecordedRequests(request());
+
+              // verify metrics
+              List<Metric> metrics = extractMetricsFromRequests(requests);
+              assertThat(metrics)
+                  .extracting(Metric::getName)
+                  .contains("histogram-test-otlp-exporter", "http.server.duration");
+            });
+  }
```

Comment:
should this verification be in a specific smoke test (instead of under the framework)?